pkg/framework/guest.go

```diff
@@ -105,32 +106,63 @@ func (g *Guest) Setup() error {
 }
 
 func (g *Guest) WaitForAPIDown() error {
-	apiDown := func() error {
-		_, err := g.k8sClient.
-			CoreV1().
-			Services("default").
-			Get("kubernetes", metav1.GetOptions{})
-
-		if err == nil {
-			return microerror.Mask(fmt.Errorf("API up"))
+	time.Sleep(1 * time.Second)
+
+	log.Println("level", "debug", "message", "waiting for k8s API to be down")
+
+	o := func() error {
+		_, err := g.k8sClient.CoreV1().Services("default").Get("kubernetes", metav1.GetOptions{})
+		if err != nil {
```

Comment:
You could check for the actual error you expect here, that way the 1 second wait isn't needed. This would also prevent this from passing because something is actually wrong with the API, or is that intended?

Reply:
This is actually how it was before just the other way around. When you tell me which error to match against then we can improve this. The thing is that there is nothing like a "this thing is not up" error. It might be an url error, or net error or timeout error or whatever.

Reply:
I see the issue, still not very satisfying I guess.